driftbase/match_placements.py

```diff
@@ -22,36 +24,108 @@
+            if expected_match_placement_id else "No match placement found"
         raise NotFoundException(message)
 
     if expected_match_placement_id and expected_match_placement_id != placement_id:
-        log.warning(f"Player '{player_id}' attempted to fetch match placement '{expected_match_placement_id}', but the player didn't issue the match placement")
-        raise UnauthorizedException(f"You don't have permission to access match placement {expected_match_placement_id}")
+        log.warning(f"Player '{player_id}' attempted to fetch match placement '{expected_match_placement_id}'"
+                    f", but the player didn't issue the match placement")
+        raise UnauthorizedException(f"You don't have permission to access match placement"
+                                    f" {expected_match_placement_id}")
 
     with JsonLock(_get_match_placement_key(placement_id)) as match_placement_lock:
         if placement_id != g.redis.conn.get(player_match_placement_key):
-            log.warning(f"Player '{player_id}' attempted to get match placement '{placement_id}', but was no longer assigned to the match placement after getting the lock")
+            log.warning(f"Player '{player_id}' attempted to get match placement '{placement_id}'"
+                        f", but was no longer assigned to the match placement after getting the lock")
             raise ConflictException("You were no longer assigned to the match placement while attempting to fetch it")
 
         placement = match_placement_lock.value
 
         if not placement:
-            log.warning(f"Player '{player_id}' is assigned to match placement '{placement_id}' but the match placement doesn't exist")
+            log.warning(f"Player '{player_id}' is assigned to match placement '{placement_id}'"
+                        f" but the match placement doesn't exist")
             g.redis.conn.delete(player_match_placement_key)
             raise NotFoundException("No match placement found")
 
         log.info(f"Returning match placement '{placement_id}' for player '{player_id}'")
 
         return placement
 
+
+def add_player_to_public_match_placement(player_id: int, placement_id: str) -> typing.Union[dict, None]:
+    with JsonLock(_get_match_placement_key(placement_id)) as match_placement_lock:
+        placement = match_placement_lock.value
+        if not placement:
+            log.warning(f"Player '{player_id}' attempted to join match placement '{placement_id}'"
+                        f" but the match placement doesn't exist")
+            raise NotFoundException("Match placement not found")
+
+        if not placement.get("public"):
+            log.warning(f"Player '{player_id}' attempted to join match placement '{placement_id}'"
+                        f" but the match placement is not public")
+            raise UnauthorizedException("You can't join a public match placement")
```

Comment:
Should probably be 403 Forbidden instead of 401 Unauthorized in this case, since the client identity is known